Add support for custom price fields - resolve price refence

junajan opened the pull request:
Fix for https://github.com/sphereio/sphere-product-import/issues/87

hajoeichler commented:
@Siilwyn and @hisabimbola can you please review too

hajoeichler commented:
@junajan can you first ensure the tests are green on travis.

junajan commented:
@hajoeichler  - Project used for travis tests (`importertest-42`) has expired. I already asked someone to activate it.

hajoeichler commented:
The project is active.

butenkor commented on test/integration.spec.coffee:
```diff
@@ -328,8 +330,72 @@ describe 'Product import integration tests', ->
-
+      expect(@import._summary.failed).toBe 1
+      expect(@import._errors.length).toBe 1
+      expect(@import._errors[0]).toBe expectedError
```
If you want to test error message you might add optional flag to product-import constructor called ```failOnError``` and then test against rejected value?

junajan replied:
Updated here 61413cd

panshul007 commented:
@junajan please add the new option of providing the error call back to the documentation. if possible with a reference to the test example.

junajan commented:
@panshul007 @butenkor please check this commit 789326a
I updated readme and the way how logging works.

junajan commented:
@butenkor Check this update please 580441e

I added `@logger` service as another parameter to the errorCallback function - I think it won't harm anything. 
I also added better validation of `errorCallback` variable.
